packages/host/app/components/operator-mode/container.gts

```diff
@@ -136,6 +139,65 @@ export default class OperatorModeContainer extends Component<Signature> {
     return await this.operatorModeStateService.constructRecentCards();
   });
 
+  @action async onCommand(command: any) {
+    // apply patch
+    if (command.type == 'patch') {
+      for (let item of this.allStackItems) {
+        if ('card' in item && item.card.id == command.id) {
+          await this.setFieldValues(item.card, command.patch.attributes);
+          return await this.save(item);
+        }
+      }
+    }
+  }
+
+  @action async onPreviewCommand(command: any) {
+    // apply patch
+    if (command.type == 'patch') {
+      for (let item of this.allStackItems) {
+        if ('card' in item && item.card.id == command.id) {
+          await this.saveCardFieldValues(item.card);
+          await this.setFieldValues(item.card, command.patch.attributes);
+        }
+      }
+    }
+  }
+
+  @action async onCancelPreviewCommand(command: any) {
+    // reset field values
+    if (command.type == 'patch') {
+      for (let item of this.allStackItems) {
+        if ('card' in item && item.card.id == command.id) {
+          return await this.cancel(item);
+        }
+      }
+    }
+  }
+
+  private async setFieldValues(card: Card, values: any) {
+    let fields = await this.cardService.getFields(card);
+    for (let fieldName of Object.keys(values)) {
+      let field = fields[fieldName];
+      if (fieldName === 'id') continue;
+      try {
+        if (
+          (field.fieldType === 'contains' ||
+            field.fieldType === 'containsMany') &&
+          !(await this.cardService.isPrimitive(field.card))
+        ) {
+          await this.setFieldValues(
+            (card as any)[fieldName],
+            values[fieldName]
+          );
+        } else {
+          (card as any)[fieldName] = values[fieldName];
```

Comment:
this runs the card instance's property descriptor's set function, which in turn excutes any card change subscriber callbacks, which in turn will perform auto saves on any card instances that have been loaded in the stack item component